Tests/Tracking/CustomerIOIntegrationTests.swift

```diff
@@ -156,4 +156,11 @@ class CustomerIOIntegrationTests: IntegrationTest {
         XCTAssertEqual(CustomerIO.shared.config!.siteId, givenSiteId)
         XCTAssertEqual(CustomerIO.shared.config!.apiKey, givenApiKey)
     }
+
+    // Note: It's very important that this is an integration test that uses *real* instances of ThreadUtil to run code in a real background thread and the dependencies in SDK initialization, such as CleanupRepository, are all real instances.
+    func test_initializeSdk_givenCallMultipleTimes_expectNoExceptions() {
```

Comment:
can we have a test for multiple threads as well maybe using dispatch queue?

Reply:
That is a great suggestion. 

[From this comment](https://github.com/customerio/customerio-ios/pull/399/files#r1385062478), I tried to explain that there was a miscommunication with what this bug is. 

Based on that, this test is a good test that reproduces the crash stacktrace reported to us from customers.

Reply:
Awesome, thanks for clarifying that. 🙏